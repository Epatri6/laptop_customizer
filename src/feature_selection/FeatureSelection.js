import React from 'react';
// Normalizes string as a slug - a string that is safe to use
// in both URLs and html attributes
import slugify from 'slugify';
import './FeatureSelection.css';
import FeatureItem from '../feature_item/FeatureItem';

export default class FeatureSelection extends React.Component {

  render() {
    const {id, feature, featureName, selectedFeature, updateFeature} = this.props;
    const options = feature.map((item) => {
      const itemHash = slugify(JSON.stringify(item));
      return <FeatureItem key={itemHash} id={itemHash} item={item} featureName={featureName} selectedFeature={selectedFeature} updateFeature={updateFeature}/>;
    });

    return (
        <fieldset className="feature" key={id}>
          <legend className="feature__name">
            <h3>{featureName}</h3>
          </legend>
          {options}
        </fieldset>
    );
  }

  static defaultProps = {
    id: '',
    feature: [],
    featureName: '',
    selectedFeature: {},
    updateFeature: () => {},
  }

}