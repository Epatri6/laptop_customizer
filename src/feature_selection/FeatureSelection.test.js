import React from 'react';
import ReactDOM from 'react-dom';
import FeatureSelection from './FeatureSelection';
import renderer from 'react-test-renderer';

describe('Testing ShoppingCart Component', () => {

  const testProps = {
    id: '1',
    feature: [
      {
        name: '17th Generation Intel Core HB (7 Core with donut spare)',
        cost: 700
      },
      {
        name: 'Professor X AMD Fire Breather with sidewinder technology',
        cost: 1200
      }
    ],
    featureName: 'Processor',
    selectedFeature: {
      name: '17th Generation Intel Core HB (7 Core with donut spare)',
      cost: 700
    },
    updateFeature: () => {},
  };

  const {id, feature, featureName, selectedFeature, updateFeature} = testProps;

  //Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FeatureSelection id={id} feature={feature} featureName={featureName} selectedFeature={selectedFeature} updateFeature={updateFeature} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FeatureSelection />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  it('renders as expected', () => {
    const tree = renderer
      .create(<FeatureSelection id={id} feature={feature} featureName={featureName} selectedFeature={selectedFeature} updateFeature={updateFeature} />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
  
  it('renders as expected without props', () => {
    const tree = renderer
      .create(<FeatureSelection />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});