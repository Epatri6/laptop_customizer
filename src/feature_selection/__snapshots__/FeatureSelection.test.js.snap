// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ShoppingCart Component renders as expected 1`] = `
<fieldset
  className="feature"
>
  <legend
    className="feature__name"
  >
    <h3>
      Processor
    </h3>
  </legend>
  <div
    className="feature__item"
  >
    <input
      checked={true}
      className="feature__option"
      id="\\"name\\":\\"17th-Generation-Intel-Core-HB-(7-Core-with-donut-spare)\\"\\"cost\\":700"
      name="Processor"
      onChange={[Function]}
      type="radio"
    />
    <label
      className="feature__label"
      htmlFor="\\"name\\":\\"17th-Generation-Intel-Core-HB-(7-Core-with-donut-spare)\\"\\"cost\\":700"
    >
      17th Generation Intel Core HB (7 Core with donut spare)
       (
      $700.00
      )
    </label>
  </div>
  <div
    className="feature__item"
  >
    <input
      checked={false}
      className="feature__option"
      id="\\"name\\":\\"Professor-X-AMD-Fire-Breather-with-sidewinder-technology\\"\\"cost\\":1200"
      name="Processor"
      onChange={[Function]}
      type="radio"
    />
    <label
      className="feature__label"
      htmlFor="\\"name\\":\\"Professor-X-AMD-Fire-Breather-with-sidewinder-technology\\"\\"cost\\":1200"
    >
      Professor X AMD Fire Breather with sidewinder technology
       (
      $1,200.00
      )
    </label>
  </div>
</fieldset>
`;

exports[`Testing ShoppingCart Component renders as expected without props 1`] = `
<fieldset
  className="feature"
>
  <legend
    className="feature__name"
  >
    <h3>
      
    </h3>
  </legend>
</fieldset>
`;
